ark/utils/spatial_analysis_utils.py

```diff
@@ -380,3 +382,35 @@ def compute_neighbor_counts(current_fov_neighborhood_data, dist_matrix, distlim,
     freqs = counts.T / num_neighbors
 
     return counts, freqs.T
+
+
+def cluster_neighborhood_matrix(neighbor_mat_data, max_k=10):
+    """For a given neighborhood matrix, cluster and compute metric scores. Uses k-means clustering.
+    Currently only supporting silhouette score as a cluster metric.
+
+    Args:
+        neighbor_mat_data (pandas.DataFrame):
+            neighborhood matrix data with only the desired fovs
+        max_k (int):
+            the maximum k we want to generate cluster statistics for, must be at least 2
+
+    Returns:
+        xarray.DataArray:
+            contains a single dimension, cluster_num, which determines the metric score
+            when cluster_num was set as k for k-means clustering
+    """
+
+    # create array we can store the results of each k for clustering
+    coords = [np.arange(2, max_k + 1)]
+    dims = ["cluster_num"]
+    stats_raw_data = np.zeros(len(np.arange(2, max_k + 1)))
+    cluster_stats = xr.DataArray(stats_raw_data, coords=coords, dims=dims)
```

Comment:
is there a reason this is an xarray and not just a dataframe?

Reply:
It was what first came to mind since we're not indexing from 0 but from 2 (b/c that's the bare minimum you can choose to cluster). A dataframe would work too.